src/libotutil/ot-keyfile-utils.c

```diff
@@ -60,6 +60,62 @@ ot_keyfile_get_boolean_with_default (GKeyFile *keyfile, const char *section, con
   return TRUE;
 }
 
+static gint
+strcmp_sized (const gchar *s1, size_t len1, const gchar *s2)
+{
+  size_t len2 = strlen (s2);
+  return strncmp (s1, s2, MAX (len1, len2));
+}
+
+gboolean
+ot_keyfile_get_tristate_with_default (GKeyFile *keyfile, const char *section, const char *value,
+                                      OtTristate default_value, OtTristate *out_tri, GError **error)
+{
+  g_return_val_if_fail (keyfile != NULL, FALSE);
+  g_return_val_if_fail (section != NULL, FALSE);
+  g_return_val_if_fail (value != NULL, FALSE);
+  gint i, length = 0;
+
+  GError *temp_error = NULL;
+  g_autofree char *ret_value = g_key_file_get_value (keyfile, section, value, &temp_error);
+  if (temp_error)
+    {
+      if (is_notfound (temp_error))
+        {
+          g_clear_error (&temp_error);
+          g_assert (ret_value == NULL);
+          *out_tri = default_value;
+          return TRUE;
+        }
+
+      g_propagate_error (error, temp_error);
+      return FALSE;
+    }
+
+  /* Count the number of non-whitespace characters */
+  for (i = 0; ret_value[i]; i++)
+    if (!g_ascii_isspace (ret_value[i]))
+      length = i + 1;
```

Comment:
Should this use `g_strstrip()` instead?

Reply:
I just copied what parse_boolean() did, but strstrip seems useful.